src/main/scala/io/iohk/ethereum/network/p2p/messages/BaseETH6XMessages.scala

```diff
@@ -136,24 +154,49 @@ object BaseETH6XMessages {
 
   object SignedTransactions {
 
-    lazy val chainId: Byte = Config.blockchains.blockchainConfig.chainId
+    lazy val chainId: Byte = 1.toByte //Config.blockchains.blockchainConfig.chainId
 
     implicit class SignedTransactionEnc(val signedTx: SignedTransaction) extends RLPSerializable {
+
+      override def toBytes: Array[Byte] =
+        signedTx.tx match {
+          // transaction with access list encoding is defined in eip2930
+          case _: TransactionWithAccessList => Transaction.Type01 +: (super.toBytes: Array[Byte])
+          case _: LegacyTransaction         => super.toBytes
+        }
+
       override def toRLPEncodable: RLPEncodeable = {
         val receivingAddressBytes = signedTx.tx.receivingAddress
           .map(_.toArray)
           .getOrElse(Array.emptyByteArray)
-        RLPList(
-          signedTx.tx.nonce,
-          signedTx.tx.gasPrice,
-          signedTx.tx.gasLimit,
-          receivingAddressBytes,
-          signedTx.tx.value,
-          signedTx.tx.payload,
-          signedTx.signature.v,
-          signedTx.signature.r,
-          signedTx.signature.s
-        )
+        signedTx.tx match {
+          case TransactionWithAccessList(nonce, gasPrice, gasLimit, _, value, payload, accessList) =>
+            RLPList(
+              chainId, // TODO improve how chainid is preserved in transactions
```

Comment:
Just a node about the `TODO`s. I see a few. Are these related with the follow-up tickets already? We already have a lot of TODO in the source code that are quite old, without a proper ticket these TODO's are never followed-up

Reply:
yes, those TODO have be put by Dom to let me know where to do stuff.
The chainId related ones will be handled by the ticket relative to the signing process.